flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/FieldStatsConverters.java

```diff
@@ -33,12 +33,14 @@
 public class FieldStatsConverters {
     private final Function<Long, List<DataField>> schemaFields;
     private final long tableSchemaId;
+    private final List<DataField> tableDataFields;
     private final ConcurrentMap<Long, FieldStatsArraySerializer> serializers;
     private final AtomicReference<List<DataField>> tableFields;
 
     public FieldStatsConverters(Function<Long, List<DataField>> schemaFields, long tableSchemaId) {
         this.schemaFields = schemaFields;
         this.tableSchemaId = tableSchemaId;
```

Comment:
We can remove tableSchemaId here if we get table data fields in the constructor

Reply:
Still have check below: `if (tableSchemaId == id)`